addons/edi/models/ir_attachment.py

```diff
@@ -0,0 +1,31 @@
+from odoo import api, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+
+import fnmatch
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    @api.constrains("res_model")
+    def check_filename(self):
+        """Validates attachment filename against the format defined in 'glob'"""
+        EdiDocument = self.env["edi.document"]
+        EdiGatewayPath = self.env["edi.gateway.path"]
+        self.ensure_one()
```

Comment:
Instead of self.ensure_one() we can do something like:
```
edi_document_attachments = self.filtered(lambda a: a.res_model == "edi.document")
for attachment in edi_document_attachments:
   doc = EdiDocument.browse(attachment.res_id)
   ...
```